src/dist/replication/meta_server/meta_split_service.cpp

```diff
@@ -131,5 +131,145 @@ void meta_split_service::do_app_partition_split(std::shared_ptr<app_state> app,
     _meta_svc->get_meta_storage()->set_data(
         _state->get_app_path(*app), std::move(value), on_write_storage_complete);
 }
+
+void meta_split_service::register_child_on_meta(register_child_rpc rpc)
+{
+    const auto &request = rpc.request();
+    auto &response = rpc.response();
+    response.err = ERR_IO_PENDING;
+
+    zauto_write_lock(app_lock());
+    std::shared_ptr<app_state> app = _state->get_app(request.app.app_id);
+    dassert_f(app != nullptr, "app is not existed, id({})", request.app.app_id);
+    dassert_f(app->is_stateful, "app is stateless currently, id({})", request.app.app_id);
+
+    dsn::gpid parent_gpid = request.parent_config.pid;
+    dsn::gpid child_gpid = request.child_config.pid;
+    partition_configuration parent_config = app->partitions[parent_gpid.get_partition_index()];
+    partition_configuration child_config = app->partitions[child_gpid.get_partition_index()];
+    config_context &parent_context = app->helpers->contexts[parent_gpid.get_partition_index()];
+
+    if (request.parent_config.ballot < parent_config.ballot) {
+        dwarn_f("partition({}) register child failed, request is out-dated, request ballot = {}, "
+                "meta ballot = {}",
+                parent_gpid,
+                request.parent_config.ballot,
+                parent_config.ballot);
+        response.err = ERR_INVALID_VERSION;
+        return;
+    }
+
+    if (child_config.ballot != invalid_ballot) {
```

Comment:
Is the new `child_config` added to `app->partitions` by L149 ? What the initialize value of `ballot` would be ?

Reply:
Child partition_configuration structure is initialized when client sent partition split request to meta server, and init value of child ballot is `invalid_ballot`(=-1). The related codes can be found in function `do_app_partition_split`(L98 ~ L133) in `meta_split_service.cpp`.